### ISSUE TITLE
Consider changing extended: true for urlencoded parser to be W3C JSON forms

### ISSUE THREAD

**Issue by dougwilson** [expressjs/body-parser#72]
Working draft: http://www.w3.org/TR/html-json-forms/
Thoughts from `qs`: https://github.com/hapijs/qs/issues/63

This support may not even be called `extended: true` within the `urlencoded` parser; it may move to be an option of the JSON parser.

**Comment by dougwilson** [comment]
Since `qs` will not be going this direction, we will be creating a new module and switching to that with 2.0

**Comment by jonathanong** [comment]
> Since qs will not be going this direction, we will be creating a new module and switching to that with 2.0

:+1:

**Comment by LinusU** [comment]
With the release of `1.0` of [multer](https://github.com/expressjs/multer) we switched to be W3C JSON forms compatible. The logic is contained in a separate module called [append-field](https://github.com/LinusU/node-append-field) which is an implementation of the algorithm described in the spec, with tests to see that it works as well.

Note that this is not a querystring parser per se, but it would be extremely easy to write one on top of that. If you want one I'll write it for you :+1:

**Comment by LinusU** [comment]
It's also very easy to write it so that it operates on a stream instead of a buffer which I think would be an advantage. That's how it works in multer`, hence the api is `appendField(strore, key, val)` which is meant to be called once for every key-value pair.

**Comment by MarkHerhold** [comment]
qs is discussing the future of the module here: https://github.com/hapijs/qs/issues/140

TL;DR It seems the trend it heading to EOL qs.

**Comment by jonchurch** [comment]
The W3C html-json-forms spec was abandoned. Closing
